API-Reference/Macros.md

```diff
@@ -0,0 +1,130 @@
+# Macros
+
+## Version Macros
+
+| Macro | Description |
+| ----- | ----------- |
+| `KOKKOS_VERSION` | The Kokkos version; `KOKKOS_VERSION % 100` is the patch level, `KOKKOS_VERSION / 100 % 100` is the minor version, and `KOKKOS_VERSION / 10000` is the major version. |
+
+## General Settings
+
+| Macro | Description |
+| ------| ---------------|
+| `KOKKOS_ENABLE_DEBUG` | Defined if extra debug features are activated. |
+| `KOKKOS_ENABLE_DEBUG_BOUNDS_CHECK` | Defined if extra bounds checks are enabled. |
+| `KOKKOS_ENABLE_DEBUG_DUALVIEW_MODIFY_CHECK` | Defined if debug checks for `Kokkos::DualView` objects are enabled. |
+| `KOKKOS_ENABLE_DEPRECATED_CODE_3` | Defined if features deprecated in major release 3 are still available. |
+| `KOKKOS_ENABLE_DEPRECATION_WARNING` | Defined if deprecated features generate deprecation warnings. |
+| `KOKKOS_ENABLE_PROFILING_LOAD_PRINT` | Kokkos will output a message when the profiling library is loaded. |
+| `KOKKOS_ENABLE_TUNING` | Whether bindings for tunings are available (see [#2422](https://github.com/kokkos/kokkos/pull/2422)). |
+| `KOKKOS_ENABLE_COMPLEX_ALIGN` | Whether complex types are aligned. |
+| `KOKKOS_ENABLE_AGGRESSIVE_VECTORIZATION` | Whether certain dependency assumptions are ignored for aggressive vectorization of internal Kokkos loops. |
+
+## Execution Spaces
+
+The following macros can be used to test whether or not a specified execution space
+is enabled. They can be tested for existence (e.g. `#ifdef KOKKOS_ENABLE_SERIAL`).
+
+| Macro                        | Description                                                                 |
+| ---------------------------- | --------------------------------------------------------------------------- |
+| `KOKKOS_ENABLE_SERIAL`       | Defined if the [`Serial`](Kokkos%3A%3ASerial) execution space is enabled.   |
+| `KOKKOS_ENABLE_OPENMP`       | Defined if the [`OpenMP`](Kokkos%3A%3AOpenMP) execution space is enabled.   |
+| `KOKKOS_ENABLE_OPENMPTARGET` | Defined if the experimental `OpenMPTarget` execution space is enabled.      |
+| `KOKKOS_ENABLE_THREADS`      | Defined if the [`Threads`](Kokkos%3A%3AThreads) execution space is enabled. |
+| `KOKKOS_ENABLE_CUDA`         | Defined if the [`Cuda`](Kokkos%3A%3ACuda) execution space is enabled.       |
+| `KOKKOS_ENABLE_HIP`          | Defined if the experimental `HIP` execution space is enabled.               |
+| `KOKKOS_ENABLE_HPX`          | Defined if the [`HPX`](Kokkos%3A%3AHPX) execution space is enabled.         |
+| `KOKKOS_ENABLE_MEMKIND`      | Defined if the experimental `HBWSpace` execution space is enabled.          |
```

Comment:
This is not an execution space is it?

Reply:
No, it's a memory space.

Reply:
So plz fix the comment.

Reply:
8c0c54492b73bb11fed8f4eaf9fdd74b87960469

Reply:
Thanks for fixing, wasn't 100% sure how it was supposed to be documented.